FreeRTOS-Plus/Demo/FreeRTOS_Plus_WolfSSL_FIPS_Ready_Windows_Simulator/README_wolfSSL_FIPS_ready.md

```diff
@@ -0,0 +1,91 @@
+
+This demo shows that wolfSSL FIPS Ready provides a FIPS compliant cryptographic module with minimal impact on client code. 
+
+# What does FIPS 140-2 specify?
+
+FIPS 140-2 enforces cryptographic modules to follow best practices, including:
+
+1. Removal of insecure algorithms (such as MD5 and DES)
+2. Include a default entry point
+3. Perform a Power On Self Test (POST)
+
+wolfSSL FIPS Ready fulfils these requirements. The third requirement means that the POST should run automatically whenever the application using the FIPS code starts up. For wolfSSL FIPS Ready, the POST consits of two tests: 
+
+- In-Core Integrity Check (HMAC-SHA256 over cryptographic algorithm object files)
+- Known Answer Tests (KATs)
+
+The in-core integrity check performs an HMAC-SHA256 operation over the object files included in the FIPS-compliant algorithm boundary. The cryptographic boundary is the FIPS-specific code and its related static data in the memory of the program. In the integrity check process, the calculated hash value is compared with the expexted pre-calculated value in the memory. Failure of this check means that compiled boundary code was modified after it was compiled. If either the integrity check or KAT fails, the module enters an error state.
+
+The KAT (Known Answer Tests) run algorithm test cases using pre-computed NIST test vectors, thus verifying that the algorithms are working successfully. The KAT code and test vectors are inside the cryptographic boundary and are also checked as part of the in-core integrity check.
+
+# How to build and run the Demo application
+
+By double-clicking the solution file named "FreeRTOS_Plus_WolfSSL_FIPS_Ready.sln" included in this folder, Visual Studio starts and shows you a project in its solution explorer. It is named "RTOSDemo" and provides a console application program which runs on windows. 
+
+All required settings have been set in the user_settings.h header file included in the RTOSDemo/FreeRTOS+/wolfSSL folder in the solution explorer pane. The next step is to build the RTOSDemo application:
+
+1. Build the RTOSDemo project
+2. Run the RTOSDemo.exe 
+
+You will see a console that pops up, and it shows output like the following:
+
+```
+Starting Power On Self Test
+In core integrity check error: hash = C66491A040D5B9686BAA7A75A280290D91B49...
+ERROR: -203
+Power On Self Test(Known-Answer-Test) FAILURE
+```
+
+Do not warry about this result, an error is expected at this point. Error number "-203" means In-Core-integrity-check failed. The check is identical to the "In Core Integrity Test" listed in the previous section. And the subsequent KAT also failed due to the first error. Once FIPS Ready has failed POST, it enters an error state and never allows subsequent cryptographic operations until the device is restarted and the tests can complete successfully. 

```

Comment:
This snippet might be useful to transcribe to the demo in a condensed form. Right now the demo is kind of bare, and users might miss this info. Perhaps giving this info a header in the README could be useful as well,

Eg. "#Self Test Explanation

Reply:
Or to just point the user to the README, if the output is confusing

Reply:
Thank you for your advice. Added POST section in this README in 448d2c.